### PR TITLE
Update sub millisecond transaction timeout handling

### PR DESCRIPTION
**Pull request by injectives**
If transaction timeout value has a sub millisecond value, it gets rounded up to next millisecond value.

### CODE REVIEW

File: driver/src/main/java/org/neo4j/driver/internal/messaging/request/TransactionMetadataBuilder.java [hunk 1]
```diff
@@ -87,7 +78,14 @@ public static Map<String, Value> buildMetadata(
             result.put(BOOKMARKS_METADATA_KEY, value(bookmarks.stream().map(Bookmark::value)));
         }
         if (txTimeoutPresent) {
-            result.put(TX_TIMEOUT_METADATA_KEY, value(txTimeout.toMillis()));
+            var millis = txTimeout.toMillis();
+            if (txTimeout.toNanosPart() % 1_000_000 > 0) {
```
Comment (thelonelyvulpes) [inline driver/src/main/java/org/neo4j/driver/internal/messaging/request/TransactionMetadataBuilder.java:82]: Another option for this check is to muliply the millis and check against nanos, mod 1,000,000 is probably clearer though.
```suggestion
            if (txTimeout.toNanosPart() != millis * 1_000_000) {
```
This validation could be in some method `checkForSubmillisecondPrecision` too, just to document for yourself in the future.

**Review by thelonelyvulpes — APPROVED** [review]
Just some ideas, looks good.

File: driver/src/test/java/org/neo4j/driver/internal/messaging/request/TransactionMetadataBuilderTest.java [hunk 2]
```diff
@@ -119,11 +127,44 @@ void shouldIncludeNotificationConfig() {
...
 
         var expectedMetadata = new HashMap<String, Value>();
         expectedMetadata.put("notifications_minimum_severity", value("WARNING"));
         expectedMetadata.put("notifications_disabled_categories", value(Set.of("UNSUPPORTED")));
         assertEquals(expectedMetadata, metadata);
     }
+
+    @ParameterizedTest
+    @ValueSource(longs = {1, 1_000_001, 100_500_000, 100_700_000, 1_000_000_001})
+    void shouldRoundUpFractionalTimeoutAndLog(long nanosValue) {
```
Comment (bigmontz) [inline driver/src/test/java/org/neo4j/driver/internal/messaging/request/TransactionMetadataBuilderTest.java:141]: Might also test if the round up doesn't happen and nothing is logged when the value is not fractional

**Review by bigmontz — APPROVED** [review]
⏲️

File: driver/src/test/java/org/neo4j/driver/internal/messaging/request/TransactionMetadataBuilderTest.java (hunk 2 above)
Comment (injectives) [reply driver/src/test/java/org/neo4j/driver/internal/messaging/request/TransactionMetadataBuilderTest.java:141]: That bit is technically covered by the existing `shouldHaveCorrectMetadata` test. But I guess we can have a dedicated test as well.

**Pull request merged**